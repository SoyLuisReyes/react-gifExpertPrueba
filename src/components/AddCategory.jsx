import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setinputValue] = useState()

  const onInputChange = ({target}) => {
    // console.log(target.value);
    setinputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();
    console.log(inputValue);

    if(inputValue.trim().length <=1 ) return;

    // setCategories( categories  => [inputValue, ...categories]);
    setinputValue('');
    onNewCategory(inputValue.trim());
    
    //Callback el de arriba
  }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
          <input  
            type="text" 
            placeholder="buscar.."
            value={inputValue}
            onChange={onInputChange}
          />
    </form>
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}