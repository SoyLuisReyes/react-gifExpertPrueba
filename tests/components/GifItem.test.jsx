import { render, screen  } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem.jsx', () => { 
    test('debe hacer match en el snapshot', () => { 
       
        const title = 'Darth Vader';
        const url = 'https://starwars.com/milenare.gif'

       const { container } =  render( <GifItem title={ title } url={url}/> );
       expect(container).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el url y el alt indicado', () => { 
        const title = 'Darth Vader';
        const url = 'https://starwars.com/milenare.gif'
        render( <GifItem title={title} url={url}/> );
        // screen.debug();
        const {src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

    })
})