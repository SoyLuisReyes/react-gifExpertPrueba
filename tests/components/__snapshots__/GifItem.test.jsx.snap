// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem.jsx debe hacer match en el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Darth Vader"
      src="https://starwars.com/milenare.gif"
    />
    <p>
      Darth Vader
    </p>
  </div>
</div>
`;
